### PR TITLE
Add links and info about testing failure states and output filters.

### PR DESCRIPTION
**Pull request by jmchilton**
xref https://github.com/galaxyproject/tools-iuc/pull/883 - goal is write tests to verify tools like this properly filter.

### CODE REVIEW

File: docs/best_practices/tool_xml.rst [hunk 1]
```diff
@@ -18,7 +18,7 @@ attribute of the XML tool element, e.g.
 
     <tool id="rgTF" name="Tool Factory" version="1.11">
 
-and should be incremented with each change of the wrapper that is
+sand should be incremented with each change of the wrapper that is
```
Comment (peterjc) [inline docs/best_practices/tool_xml.rst]: typo

Comment (jmchilton) [inline docs/best_practices/tool_xml.rst]: Opps - not sure how that happened. Will rebase without that.

**Comment by nsoranzo** [comment]
:+1:

**Review by yhoogstrate — CHANGES_REQUESTED** [review]
`https://`, further approved 👍

**Comment by jmchilton** [comment]
Updated to use https - thanks @yhoogstrate !

**Review by yhoogstrate — APPROVED** [review]

**Pull request merged**